### PR TITLE
fix[react test renderer]  replace deprecated type ReactType

### PR DESCRIPTION
**Pull request by maddhruv**
Replaces the deprecated `ReactType` with `ElementType` in `react-test-renderer`
Ref - https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L75

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>

### CODE REVIEW

File: types/react/index.d.ts [hunk 1]
```diff
@@ -320,7 +321,7 @@ declare namespace React {
         /**
          * **NOTE**: Exotic components are not callable.
          */
-        (props: P): (ReactElement|null);
```
Comment (eps1lon) [inline types/react/index.d.ts]: This seems unrelated. Probably branched off from another branch instead of `master`?

**Review by eps1lon — CHANGES_REQUESTED** [review]
Thanks! This is fine apart from the rogue commit.

File: types/react/index.d.ts (hunk 1 above)
Comment (maddhruv) [reply types/react/index.d.ts]: no it is from master only

Comment (maddhruv) [reply types/react/index.d.ts]: Fixed, it mistakenly got cut off from a different branch

**Review by eps1lon — APPROVED** [review]

**Comment by uniqueiniquity** [comment]
Congratulations on your first DefinitelyTyped contribution!
Thank you for being a part of the community!
